src/ast.ts

```diff
@@ -828,6 +829,7 @@ export abstract class Node {
     stmt.range = range;
     stmt.flags = flags;
     stmt.name = name; name.parent = stmt;
+    stmt.definiteAssignmentAssertion = definiteAssignmentAssertion;
```

Comment:
This could also become a part of `flags`, maybe as `CommonFlags.DEFINITE_ASSIGNMENT`, similar to other modifiers. The advantage of a flag is that it doesn't occupy additional space in the class layout. Wdyt?